lib/java-extras/src/main/java/org/triplea/io/ZipExtractor.java

```diff
@@ -61,37 +71,36 @@ public FileSystemException(final String message, final Exception e) {
    * @throws ZipReadException Thrown if the target zip file is invalid and could not be read.
    * @throws FileSystemException Thrown if there is an error during extraction.
    */
-  public static void unzipFile(final File fileZip, final File destDir)
+  public static void unzipFile(final Path fileZip, final Path destDir)
       throws ZipReadException, FileSystemException {
     Preconditions.checkArgument(
-        fileZip.getName().endsWith(".zip"),
-        "Illegal arg, must be a zip file: " + fileZip.getAbsolutePath());
-    if (destDir.exists()) {
+        fileZip.toString().endsWith(".zip"),
+        "Illegal arg, must be a zip file: " + fileZip.toAbsolutePath());
+    if (Files.exists(destDir)) {
       Preconditions.checkArgument(
-          destDir.isDirectory(),
-          "Illegal arg, destination directory must be a directory: " + destDir.getAbsolutePath());
+          Files.isDirectory(destDir),
+          "Illegal arg, destination directory must be a directory: " + destDir.toAbsolutePath());
     } else {
-      Preconditions.checkState(
-          destDir.mkdirs(),
-          "Error, was not able to create directory: " + destDir.getAbsolutePath());
+      try {
+        Files.createDirectories(destDir);
+      } catch (final IOException e) {
+        throw new IllegalStateException(
+            "Error, was not able to create directory: " + destDir.toAbsolutePath(), e);
+      }
     }
 
     // iterate over each zip entry and write to a corresponding file
-    try (ZipFile zipFile = new ZipFile(fileZip)) {
-      for (final ZipEntry zipEntry : zipFile.stream().collect(Collectors.toList())) {
-        try {
-          unzipZipEntry(destDir, zipFile, zipEntry);
-        } catch (final IOException e) {
-          throw new FileSystemException(fileZip, e);
+    try (FileSystem zipFileSystem = FileSystems.newFileSystem(fileZip, null)) {
+      final Path zipRoot = zipFileSystem.getPath("/");
+      try (Stream<Path> files = Files.walk(zipRoot)) {
```

Comment:
I incorporated my proposal from https://github.com/triplea-game/triplea/pull/8616#discussion_r555060868 into this PR.
There is a flaw with `Files#walk` that I noticed when crafting a malicious zip for testing though: If the malicious zip file uses ".." as a folder name, somewhere it is possible that this call walks forever because it recursively searches the same pair of folders forever until the heap runs out of free memory.
Obviously this doesn't happen with regular zip files, but we might want to consider limiting this to a depth of some arbitrary number like 500.

Also another thing to consider: IIRC Files#walk guarantees a fixed order the files are traversed in, so we might be able to drop the checks if the parent directories already exist because we already visited them in the past